docs/reference/commands/create-enrollment-token.asciidoc

```diff
@@ -42,12 +42,17 @@ option. For more information about debugging connection failures, see
 
 `-E <KeyValuePair>`:: Configures a standard {es} or {xpack} setting.
 
-`-f, --force`:: Forces the command to run against an unhealthy cluster. 
+`-f, --force`:: Forces the command to run against an unhealthy cluster.
 
 `-h, --help`:: Returns all of the command parameters.
 
 `-s, --scope`:: Specifies the scope of the generated token. Supported values are `node` and `kibana`.
 
+`--url`:: Specifies the base URL (hostname and port) that the tool uses to submit API
+requests to {es}. The default value is determined from the settings in your
+`elasticsearch.yml` file. If `xpack.security.http.ssl.enabled`  is set to `true`,
+you must specify an HTTPS URL.
```

Comment:
Do you think it would be preferable that we drop the "If `xpack.security.http.ssl.enabled`  is set to `true`,
you must specify an HTTPS URL" scheme condition, by having the tool check for that itself (like it does for the "default URL")?

Moreover, a URL is more than the scheme and the host and port pair, and using DNS names to refer to the node might not be ideal, as it might not be included in the SAN of the cert.
Do you think it would be preferable that this option be more focused on the **ip and port that the local node can be reached at**?

Reply:
> Do you think it would be preferable that this option be more focused on the ip and port that the local node can be reached at?

I contemplated the same but in the end settled to a single option as a simpler thing, happy to discuss. 

 >  a URL is more than the scheme and the host and port pair

true but what we should be aiming for here is not naming strictness for the sake of it but a name for the option that’s understandable by (or at least explainable for ) the majority of users. I think that `url` fits the bill. 

> and using DNS names to refer to the node might not be ideal, as it might not be included in the SAN of the cert.

the idea is that if you ever need to use this parameter you do so because you _know_ that you want to talk to the node at an ip or hostname that is in the SANs but CommandLineHttpClient picks another. So you *know* what should be in the url (either a hostname or an IP address ) 

> Do you think it would be preferable that we drop the "If xpack.security.http.ssl.enabled is set to true,
you must specify an HTTPS URL" scheme condition, by having the tool check for that itself (like it does for the "default URL")?

If we change this to two parameters then yes it makes sense , we shouldn’t be asking the users for a third parameter (scheme) but I’m not 100% that ee should change. 

Do we introduce an ip and a port parameter? Do we mandate both if one is passed or have default values for the port ?

Reply:
> Do we introduce an ip and a port parameter? Do we mandate both if one is passed or have default values for the port ?

I don't have a preference. Having to specify both the ip and the port simultaneously (as is the case with this `url` parameter), as a single option is OK.